net/procfs/net_tcp.c

```diff
@@ -0,0 +1,211 @@
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+#include <stdio.h>
+#include <string.h>
+#include <debug.h>
+
+#include <arpa/inet.h>
+#include <netinet/in.h>
+
+#include <nuttx/net/netstats.h>
+
+#include "procfs/procfs.h"
+#include "tcp/tcp.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#if defined(CONFIG_NET_TCP)
+
+#define TCP_LINELEN 120
+
+ssize_t netprocfs_tcpstats(FAR struct netprocfs_file_s *priv,
+                           FAR char *buffer, size_t buflen,
+                           uint8_t domain, int *skip)
+{
+  int addrlen = (domain == PF_INET) ?
+                INET_ADDRSTRLEN : INET6_ADDRSTRLEN;
+  FAR struct tcp_conn_s *conn = NULL;
+  char remote[addrlen + 1];
+  char local[addrlen + 1];
+  int len = 0;
+  void *laddr;
+  void *raddr;
+
+  net_lock();
+
+  if (priv->offset == 0)
+    {
+      len = snprintf(buffer, buflen, "TCP sl  "
+                                     "st flg ref tmr uack nrt   "
+#if CONFIG_NET_SEND_BUFSIZE > 0
+                                     "txsz   "
+#endif
+                                     "rxsz "
+                                     "%-*s "
+                                     "%-*s\n"
+                                     ,
+                                     INET6_ADDRSTRLEN / 2,
+                                     "local_address",
+                                     INET6_ADDRSTRLEN / 2,
+                                     "remote_address"
+                                     );
+      priv->offset = 1;
+    }
+
+  local[addrlen] = '\0';
+  remote[addrlen] = '\0';
+
+  while ((conn = tcp_nextconn(conn)) != NULL)
+    {
+#if defined(CONFIG_NET_IPv4) && defined(CONFIG_NET_IPv6)
+      if (conn->domain != domain)
+        {
+          continue;
+        }
+#endif /* CONFIG_NET_IPv4 && CONFIG_NET_IPv6 */
+
+      if (++(*skip) <= priv->offset)
+        {
+          continue;
+        }
+
+      if (buflen - len < TCP_LINELEN)
+        {
+          break;
+        }
+
+#if defined(CONFIG_NET_IPv4)
+      if (domain == PF_INET)
+        {
+          laddr = &conn->u.ipv4.laddr;
+          raddr = &conn->u.ipv4.raddr;
+        }
+#endif /* CONFIG_NET_IPv4 */
+
+#if defined(CONFIG_NET_IPv6)
+      if (domain == PF_INET6)
+        {
+          laddr = &conn->u.ipv6.laddr;
+          raddr = &conn->u.ipv6.raddr;
+        }
+#endif /* CONFIG_NET_IPv6 */
+
+      len += snprintf(buffer + len, buflen - len,
+                      "    %2" PRIu8
+                      ": %02" PRIx8
+                      " %3" PRIx8 " %3" PRIu8
+                      " %3" PRIu8
+#if defined(CONFIG_NET_TCP_WRITE_BUFFERS) || \
+    defined(CONFIG_NET_TCP_WINDOW_SCALE)
+                      " %4" PRIu32
+#else
+                      " %4" PRIu16
```

Comment:
let's cast to uint32_t to avoid #ifdef/#else.

Reply:
Done